src/components/ResourceBrowser/ResourceList/AdminTerminal.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useMemo } from 'react'
+import { useParams } from 'react-router-dom'
+import { Progressing, ErrorScreenManager, Reload, useAsync } from '@devtron-labs/devtron-fe-common-lib'
+import ClusterTerminal from '../../ClusterNodes/ClusterTerminal'
+import { createGroupSelectList, filterImageList } from '../../common'
+import { createTaintsList } from '../../cluster/cluster.util'
+import { getClusterList, clusterNamespaceList } from '../../ClusterNodes/clusterNodes.service'
+import { getHostURLConfiguration } from '../../../services/service'
+import { AdminTerminalProps, URLParams } from '../Types'
+
+const AdminTerminal: React.FC<AdminTerminalProps> = ({ isSuperAdmin, updateTerminalTabUrl }: AdminTerminalProps) => {
+    const { clusterId } = useParams<URLParams>()
+
+    const [loading, data, error] = useAsync(() =>
+        Promise.all([getClusterList(), getHostURLConfiguration('DEFAULT_TERMINAL_IMAGE_LIST'), clusterNamespaceList()]),
+    )
+
+    const [detailClusterList = null, hostUrlConfig = null, _namespaceList = null] = data || []
+
+    const selectedDetailsCluster = useMemo(
+        () => detailClusterList?.result.find((cluster) => cluster.id === +clusterId) || null,
+        [detailClusterList, clusterId],
+    )
+
+    const imageList = useMemo(() => JSON.parse(hostUrlConfig?.result.value || null), [hostUrlConfig])
+
+    const namespaceList = _namespaceList?.result || null
+
+    if (loading) {
+        return (
+            <div className="h-100 node-data-container bcn-0">
+                <Progressing pageLoader />
+            </div>
+        )
+    }
+
+    if (error) {
+        const errCode = error?.errors[0]?.['code'] || error?.['code']
```

Comment:
remove ? after error

Reply:
Why are we checking for the internal error code?

Reply:
At the time of writing I hadn't seen how fetchInTime & fetchApi throws the ServerErrors. So whenever I console logged the error I saw an error. Now that I know type of ServerErrors, I have removed the call that collects the internal error code.